src/kudu/util/hdr_histogram.cc

```diff
@@ -324,6 +325,15 @@ uint64_t HdrHistogram::ValueAtPercentile(double percentile) const {
   return 0;
 }
 
+void HdrHistogram::ResetHistogram(){
```

Comment:
This function is the main concern. Essentially, 'n' number of 'HdrHistogram' objects are created for 'n' metrics in the system. This will create a lot of  'histogram_mutex_' locks and this particular function will be called 'n' times when the stats are collected. We decided to look at it later.

Reply:
My speculation is that we might not see this mutex when we run at scale. 
Few things to note:
* ResetHistogram is called by diagnostics thread which runs after fixed duration for each metrics sequentially. That thread is not on hot code path, so we are ok if hit some contention while running diagnostics thread.
* As we have discussed offline, this feature is guarded by feature flag and can be turned off if required. 

If we face any issue while running in production, we will resist this later.